security/kubernetes-nodes.md

```diff
@@ -0,0 +1,280 @@
+  # This rule allows ingress to the Kubernetes API server.
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports:
+      # kube API server
+      - 6443
+  # This rule allows traffic to Kubernetes control plane ports
+  # from localhost. The health check port for calico/node is also whitelisted
+  - action: Allow
+    protocol: TCP
+    destination:
+      nets:
+      - 127.0.0.1/32
+      ports:
+      # etcd server client API
+      - "2379:2381"
+      # kube-scheduler
+      - 10251
+      - 10259
+      # kube-controller-manager
+      - 10252
+      - 10257
+      # kubelet API
+      - 10250
+      # calico/node health check
+      - 9099
+  # This rule is required in multi-master clusters where etcd pods are colocated with the masters.
+  # Allow the etcd pods on the masters to communicate with each other. 2380 is the etcd peer port.
+  - action: Allow
+    protocol: TCP
+    source:
+      selector: has(node-role.kubernetes.io/master)
+    destination:
+      ports:
+      - 2380
+EOF
+```
+
+Note that the above policy selects the standard Kubernetes label **node-role.kubernetes.io/master** attached to master nodes.
+
+Next, we need to apply policy to restrict ingress to the Kubernetes workers.
+Before adding the policy we will add a label to all of our worker nodes, which then gets added to its automatic host endpoint.
+For this tutorial we will use **kubernetes-worker**. An example command to add the label to worker nodes:
+
+```bash
+kubectl get node -l '!node-role.kubernetes.io/master' -o custom-columns=NAME:.metadata.name | tail -n +2 | xargs -I{} kubectl label node {} kubernetes-worker=
+```
+
+The workers' ingress policy consists of two rules. As with the masters, the worker nodes need to access their kubelet API and calico/node healthcheck.
+Now apply the policy:
+
+```
+calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: workers
+spec:
+  selector: has(kubernetes-worker)
+  # Allow kubelet API access from the docker API and allow calico/node health check.
+  # An ephemeral port # is used by kubelet and docker for commands like 'kubectl exec'
+  # to a pod on a worker node.
+  # Note: This rule's port range will differ based on your system's ephemeral port range.
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      nets:
+      - 127.0.0.1/32
+      ports:
+      # kubelet API
+      - 10250
+      - "32768:60999"
+      # calico/node health check
+      - 9099
+  # Allow the masters access to the nodes kubelet API.
+  - action: Allow
+    protocol: TCP
+    source:
+      selector: has(node-role.kubernetes.io/master)
+    destination:
+      ports:
+      - 10250
+EOF
+```
+
+Now that the policies are in place, the final task is to update the **FailsafeInboundHostPorts** and remove unnecessary failsafe ports.
+Without this step, the control plane is still accessible on ports 2379 and 2380.
```

Comment:
What does control plane mean?  I think better to spell out the specific components / use of these ports?

Reply:
ef06476868e6f2c9acfbf52382187940dc79266b